core/extension_set.rs

```diff
@@ -146,12 +156,37 @@ pub fn create_op_ctxs(
       runtime_state_ptr,
       get_error_class_fn,
       metrics_fn,
-    );
+    )
+  };
 
-    op_ctxs.push(op_ctx);
+  for (index, decl) in op_decls.into_iter().enumerate() {
+    op_ctxs.push(create_ctx(index, decl));
   }
 
-  op_ctxs.into_boxed_slice()
+  for (index, mut decl) in op_method_decls.into_iter().enumerate() {
+    decl.constructor.name = decl.name.0;
+    decl.constructor.name_fast = decl.name.1;
```

Comment:
I hope that methods can't participate in the "op middleware" infra? (Maybe they should?) If they do, make sure that overriding the name in the middleware doesn't actually override it otherwise you'll get call mismatches

Reply:
nope, this doesn't participate in op middleware infra